### ISSUE TITLE
Wrong condition in checkCircularReference method

### ISSUE THREAD

**Issue by sephrouse** [kotlin-orm/ktorm#433]
There is a check method as follows in BaseTable.kt. Suppose the condition **schema == root.catalog** is a mistake. From my understanding, it should be **schema == root.schema**.

Please check out and fix it if it was wrong.

```
private fun checkCircularReference(root: BaseTable<*>, stack: LinkedList<String> = LinkedList()) {
        stack.push(root.toString(withAlias = false))

        if (tableName == root.tableName && catalog == root.catalog && schema == root.catalog) {
            throw IllegalStateException(
                "Circular reference detected, current table: '$this', reference route: ${stack.asReversed()}"
            )
        }

        for (column in root.columns) {
            val ref = column.referenceTable
            if (ref != null) {
                checkCircularReference(ref, stack)
            }
        }

        stack.pop()
    }
```

**Comment by vincentlauvlwj** [comment]
Thank you for your feedback, the bug will be fixed in the next version.

**Comment by vincentlauvlwj** [comment]
Fixed in 3.6.0
